app/controllers/uploads_controller.rb

```diff
@@ -99,4 +102,46 @@ def call_load
   def klass
     @upload.csv_type.constantize
   end
+
+  def validators_messages
+    # this a call to custom validators that are not listed inside the class
+    validator_klass = "#{klass.name}Validator".safe_constantize
+
+    messages = validation_messages
+    messages.push(*validator_klass::VALIDATION_DESCRIPTIONS) if validator_klass?(validator_klass)
+
+    messages
+  end
+
+  def validator_klass?(validator_klass)
+    # this a call to custom validators that are not listed inside the class
+    validator_klass.present? && defined? validator_klass::VALIDATION_DESCRIPTIONS
+  end
+
+  def validation_messages
+    klass.validators.map do |validations|
+      if validations.class == ActiveRecord::Validations::PresenceValidator
+        generic_validator('These columns must have a value: ', validations)
+      elsif validations.class == ActiveModel::Validations::InclusionValidator
+        inclusion_validator(validations)
+      elsif validations.class == ActiveModel::Validations::NumericalityValidator
+        generic_validator('These columns can only contain numeric values: ', validations)
+      elsif validations.class == ActiveRecord::Validations::UniquenessValidator
+        generic_validator('These columns should contain unique values: ', validations)
+      end
```

Comment:
in ruby `case` uses `===` instead of `==` which causes issues when checking if something is a specific class 
https://ruby-doc.org/docs/keywords/1.9/Object.html#method-i-case
Actually I should probably be using https://apidock.com/ruby/Object/is_a%3F

Reply:
changed